### ISSUE TITLE
Enabling plists results in `Error processing message (wrong-type-argument hash-table-p (:jsonrpc "2.0"`

### ISSUE THREAD

**Issue by altschuler** [emacs-lsp/lsp-mode#3716]
### Thank you for the bug report

- [X] I am using the latest version of `lsp-mode` related packages.
- [X] I checked [FAQ](https://emacs-lsp.github.io/lsp-mode/page/faq/) and [Troubleshooting](https://emacs-lsp.github.io/lsp-mode/page/troubleshooting/) sections
- [ ] You may also try reproduce the issue using clean environment using the following command: `M-x lsp-start-plain`

### Bug description

After enabling plists using `LSP_USE_PLISTS` and recompiling, then following error occurs when lsp attempts any communication with the language server (`ts-ls`). I have followed the instructions (set the env flag, delete lsp related package, restart, recompile), I have also tried recompiling using `byte-recompile-directory`, but with the same result.

```
Error processing message (wrong-type-argument hash-table-p (:jsonrpc "2.0" :method "window/logMessage" :params (:type 3 :message "[lspserver] Using Typescript version (user-setting) 4.8.3 from path \"/home/simon/.emacs.d/.cache/lsp/npm/typescript/lib/node_modules/typescript/lib/tsserver.js\""))).
```

There are a lot of errors like this, they all start with the `Error processing message (wrong-type-argument hash-table-p (:jsonrpc "2.0"` part.

### Steps to reproduce

Using hash-table mode:
1. set `LSP_USE_PLISTS` env var
2. delete all lsp-related packages
3. restart emacs
4. reinstall packages
5. verify that `lsp-use-plists` is non-nil
6. connect to a language server
7. observe errors logged

### Expected behavior

No errors, and that plists are in use.

### Which Language Server did you use?

ts-ls

### OS

Linux

### Anything else?

I'm trying to enable plists in an attempt to increase performance of lsp-mode. Very often (sometimes all the time), emacs hangs for 10-20 second when communicating with the language server. I'm not certain this is due to lsp-mode, could be the language server (ts-ls). But that is somewhat besides the point of this issue.

I'm using Spacemacs.

I have read #3311 and #3313, but didn't find any help.

**Comment by altschuler** [comment]
I seem to have solved it by deleting the lsp related packages, then delete the cached elpa packages manually and the `eln-cache` (not sure what it is). Completions and other actions still hangs all the time, but that's another issue :)

**Comment by artemkovalyov** [comment]
I am trying it now. That ENV flag-based approach is so inconsistent and makes config less reproducible.

**Comment by yyoncho** [comment]
> That ENV flag-based approach is so inconsistent and makes config less reproducible.

@artemkovalyov it is not that we want that - the issue is if you using native compilation it will compile lsp-mode files in separate process without running users config. Thus we have to pass somehow the new value of that flag...

**Comment by artemkovalyov** [comment]
@yyoncho, thanks a lot for you work guys. It makes Emacs much more relevant as IDE.
This `PLISTS` issue happens intermittently when upgrading my system or using Emacs on another machine and I usually forget the root cause by then.  It takes some time to recall it and build it properly. I use `exec-path-from-shell` and could see the env var in Emacs but it finally worked only after I launched it from console.

I was going to think if I can contribute there.  Maybe at least to the docs. Thanks for taking time to explain why thing are the way they are.
